src/components/QueueMessage.js

```diff
@@ -50,22 +72,66 @@ class QueueMessage extends React.Component {
       return null
     }
 
+    const handleTabKeyPress = (e, tab) => {
+      if (e.key === 'Enter' || e.key === ' ') {
+        e.stopPropagation()
+        this.onChangeTab(tab)
+      }
+    }
+
     let content
     let button
     if (editing) {
       content = (
-        <>
-          <Input
-            type="textarea"
-            name="text"
-            rows="6"
-            value={editedMessage}
-            onChange={this.onMessageChanged}
-          />
-          <FormText color="muted">
-            You can use Markdown to format this message.
-          </FormText>
-        </>
+        <Card>
+          <CardHeader>
+            <Nav card tabs>
+              <NavItem>
+                <NavLink
+                  className={classnames({
+                    active: this.state.activeTab === '1',
+                  })}
+                  onClick={() => this.onChangeTab('1')}
+                  tabIndex="0"
+                  onKeyPress={e => handleTabKeyPress(e, '1')}
+                >
+                  Edit
+                </NavLink>
+              </NavItem>
+              <NavItem>
+                <NavLink
+                  className={classnames({
+                    active: this.state.activeTab === '2',
+                  })}
+                  onClick={() => this.onChangeTab('2')}
+                  tabIndex="0"
+                  onKeyPress={e => handleTabKeyPress(e, '2')}
+                >
+                  Preview
+                </NavLink>
+              </NavItem>
+            </Nav>
+          </CardHeader>
+          <CardBody className="p-0">
+            <TabContent activeTab={this.state.activeTab}>
+              <TabPane className="m-2" tabId="1">
+                <Input
+                  type="textarea"
+                  name="text"
+                  rows="6"
+                  value={editedMessage}
+                  onChange={this.onMessageChanged}
+                />
+                <FormText color="muted">
+                  You can use Markdown to format this message.
+                </FormText>
+              </TabPane>
+              <TabPane className="m-3" tabId="2">
+                <ReactMarkdown source={editedMessage} />
```

Comment:
Can we take a page out of GitHub's book again and render "Nothing to preview" if `editedMessage` is empty?

Reply:
Taken care of this in new commit